BlackNectar/BlackNectar/SearchFilterViewController.swift

```diff
@@ -0,0 +1,343 @@
+        
+        let annotation = CustomAnnotation(name: storeName, latitude: latitude, longitude: longitude)
+        
+        return annotation
+        
+    }
+    
+}
+
+//MARK: Map View Delegate Code
+extension SearchFilterViewController {
+    
+    func mapView(_ mapView: MKMapView, viewFor annotation: MKAnnotation) -> MKAnnotationView? {
+        
+        if annotation is MKUserLocation {
+            
+            return nil
+            
+        }
+        
+        let annotation = annotation as? CustomAnnotation
+        let smallSquare = CGSize(width: 30, height: 30)
+        let button = UIButton(frame: CGRect(origin: CGPoint.zero, size: smallSquare))
+        let blackNectarPin = UIImage(named: "BlackNectarMapPin")
+        let annotationView = MKAnnotationView(annotation: annotation, reuseIdentifier: annotation?.identifier)
+        
+        button.setBackgroundImage(UIImage(named: "carIcon"), for: .normal)
+        
+        annotationView.canShowCallout = true
+        annotationView.leftCalloutAccessoryView = button
+        annotationView.image = blackNectarPin
+        
+        return annotationView
+        
+    }
+    
+}
+
+//MARK: Gets Driving Directions Code
+
+extension SearchFilterViewController {
+    
+    
+    func getDrivingDirections(to storeCoordinates: CLLocationCoordinate2D, with storeName: String) -> MKMapItem {
+        
+        let storePlacemark = MKPlacemark(coordinate: storeCoordinates, addressDictionary: [ "\(title)" : storeName ])
+        let storePin = MKMapItem(placemark: storePlacemark)
+        storePin.name = storeName
+        
+        return storePin
+        
+    }
+    
+    func mapView(_ mapView: MKMapView, annotationView view: MKAnnotationView, calloutAccessoryControlTapped control: UIControl) {
+        
+        let appleMapslaunchOptions = [MKLaunchOptionsDirectionsModeKey: MKLaunchOptionsDirectionsModeDriving]
+        
+        if let storeLocation = view.annotation {
+            
+            let storeName: String = (storeLocation.title ?? nil) ?? "Uknown"
+            
+            getDrivingDirections(to: storeLocation.coordinate, with: storeName).openInMaps(launchOptions: appleMapslaunchOptions)
+            
+            AromaClient.beginMessage(withTitle: "User tapped on \(storeName) map pin")
+                .addBody("User navigated to \(storeName)\nstore coordinates: \(storeLocation.coordinate)\n(Search Filter Map View)")
+                .withPriority(.medium)
+                .send()
+            
+        }
+        
+    }
+    
+}
+
+//MARK: Loads Stores When User Pans 
+
+extension SearchFilterViewController {
+    
+    func mapView(_ mapView: MKMapView, regionDidChangeAnimated animated: Bool) {
+        
+        let center = mapView.centerCoordinate
+        
+        LOG.debug("User dragged Map Screen to: \(center)")
+        
+        self.loadStoresInMapView(at: center)
+        
+    }
+    
+}
+
+fileprivate extension MKMapView {
```

Comment:
I'm seeing duplicate code here. Suggest this code be put somewhere so that it is only written once.